dali/kernels/signal/decibel/decibel_calculator.h

```diff
@@ -25,29 +25,36 @@ namespace signal {
 template <typename T>
 struct DecibelCalculator {
  public:
+  DALI_HOST_DEV DALI_FORCEINLINE
   explicit DecibelCalculator(T mul = 10.0, T s_ref = 1.0, T min_ratio = 1e-8)
-      : inv_mul_(1. / mul)
+      : inv_mul_(T(1) / mul)
       , mul_log2_(mul * kLog2Factor)
       , s_ref_(s_ref)
-      , inv_s_ref_(s_ref == 1.0 ? 1.0 : 1.0 / s_ref)
+      , inv_s_ref_(s_ref == T(1) ? T(1) : T(1) / s_ref)
       , min_ratio_(min_ratio) {
-    assert(min_ratio_ > 0.0);
-    assert(inv_s_ref_ > 0.0);
+    assert(min_ratio_ > T(0));
+    assert(inv_s_ref_ > T(0));
   }
 
-  DALI_FORCEINLINE
+  DALI_HOST_DEV DALI_FORCEINLINE
```

Comment:
While we're at it, I'd consider changing this from functor (`operator()`) to regular function (`signal2db`), as functor here seems not really accurate (it still requires the object to be created and has a state). But ofc it's only a suggestion, I know it would require changing the invocation in all places...

Reply:
I think this is perfectly valid case for a functor. The operator() is idempotent _and_ requires significant setup, so the object is warranted.